src/Translation_unit.ml

```diff
@@ -149,25 +149,39 @@ let parse_print (XUnit xunit) (conf : Conf.t) ~input_name ~input_file ic
   (* iterate until formatting stabilizes *)
   let rec print_check ~i ~(conf : Conf.t) ~ast ~comments ~prefix ~source_txt
       ~source_file : result =
-    let tmp, oc =
-      if not Conf.debug then Filename.open_temp_file ~temp_dir:dir base ext
+    dump xunit dir base ".old" ".ast" ast ;
+    let remaining_comments, tmp =
+      let print_and_get_remaining_comments ?(box_debug = false) oc =
+        let source = Source.create source_txt in
+        let cmts_t = xunit.init_cmts source conf ast comments in
+        let fs = Format.formatter_of_out_channel oc in
+        Fmt.set_margin conf.margin fs ;
+        (* note that [fprintf fs "%s" ""] is not a not-opt. *)
+        if not (String.is_empty prefix) then Format.fprintf fs "%s" prefix ;
+        let do_fmt = xunit.fmt source cmts_t conf ast in
+        if box_debug then Fmt.with_box_debug do_fmt fs else do_fmt fs ;
+        Format.pp_print_newline fs () ;
+        Out_channel.close oc ;
+        lazy (Cmts.remaining_comments cmts_t)
+      in
+      if not Conf.debug then
+        let tmp, oc = Filename.open_temp_file ~temp_dir:dir base ext in
+        let cmts = print_and_get_remaining_comments oc in
+        (cmts, tmp)
       else
-        let name = Format.sprintf "%s.%i%s" base i ext in
-        let tmp = Filename.concat dir name in
-        Format.eprintf "%s@\n%!" tmp ;
-        let oc = Out_channel.create ~fail_if_exists:(not Conf.debug) tmp in
-        (tmp, oc)
+        let print_to_file ?box_debug base_name =
+          let tmp = Filename.concat dir base_name in
+          Format.eprintf "%s@\n%!" tmp ;
+          let oc = Out_channel.create ~fail_if_exists:false tmp in
+          let cmts = print_and_get_remaining_comments ?box_debug oc in
+          (cmts, tmp)
+        in
+        let tmp = print_to_file (Format.sprintf "%s.%i%s" base i ext) in
+        ignore
```

Comment:
(not important) A good practice (that is not always enforced in this project) is to check for the return type of the ignored expression.

Reply:
This type is quite complicated, is `_ * _` good enough ?

Reply:
I find this syntax prettier:
```ocaml
let ( (_ : (Location.t * string * string * Sexp.t) list lazy_t)
    , (_ : string) ) =
  print_to_file ~box_debug:true
     (Format.sprintf "%s.%i.boxes%s" base i ext)
in
```
But I won't force it, maybe Josh will have an opinion about it

Reply:
I don't like this but I won't force either.

Reply:
I tend to prefer to keep all the "mess" related to `ignore` together and subordinate to the actual code, so put `|> (ignore : _ * _ -> _)` after the computation. And unless there is some particular documentation value in specifying the type more, I wouldn't do more than make sure that the computation actually gets executed.

Reply:
I did that.